services/api/src/resources/fact/resolvers.ts

```diff
@@ -139,3 +228,169 @@ export const deleteFactsFromSource: ResolverFn = async (
 
   return 'success';
 };
+
+export const addFactReference: ResolverFn = async (
+  root,
+  { input: { eid, fid, name } },
+  { sqlClientPool, hasPermission }
+) => {
+  const environment = await environmentHelpers(sqlClientPool).getEnvironmentById(eid);
+
+  await hasPermission('fact', 'add', {
+    project: environment.project
+  });
+
+  const { insertId } = await query(
+    sqlClientPool,
+    Sql.insertFactReference({
+      eid,
+      fid,
+      name
+    })
+  );
+
+  const rows = await query(
+    sqlClientPool,
+    Sql.selectFactReferenceByDatabaseId(insertId)
+  );
+
+  return R.prop(0, rows);
+};
+
+export const deleteFactReferenceById: ResolverFn = async (
```

Comment:
I'm wondering if, from a usability perspective, it might not make sense to change this slightly?

Instead of deleting by fact reference id - maybe removing a fact should just be a function of the environment id and the fact's name? There really should be no real reason for the user to know the fact reference's id, right?